app/src/test/java/com/example/trainingintive/dog_images_feature/data/repository/DogImageRepositoryImplTest.kt

```diff
@@ -0,0 +1,177 @@
+package com.example.trainingintive.dog_images_feature.data.repository
+
+import com.example.trainingintive.dog_images_feature.data.entity.DogImageUrlApiModel
+import com.example.trainingintive.dog_images_feature.data.entity.DogImageUrlDb
+import com.example.trainingintive.dog_images_feature.data.entity.toDatabaseEntity
+import com.example.trainingintive.dog_images_feature.data.local.DogImageDao
+import com.example.trainingintive.dog_images_feature.data.network.DogImageApiService
+import com.example.trainingintive.dog_images_feature.domain.model.DogImageUrl
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.unmockkStatic
+import io.mockk.verify
+import io.reactivex.rxjava3.core.Completable
+import io.reactivex.rxjava3.core.Flowable
+import io.reactivex.rxjava3.core.Single
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.util.Collections
+
+class DogImageRepositoryImplTest {
+
+    val dogApiService: DogImageApiService = mockk()
+    val dogDao: DogImageDao = mockk()
+    val tested = DogImageRepositoryImpl(dogApiService, dogDao)
+    val urlApiModel: DogImageUrlApiModel = mockk(relaxed = true)
+    val urlDomain: DogImageUrl = mockk(relaxed = true)
+
+    @Before
+    fun start() {
+        mockkStatic(DogImageUrl::toDatabaseEntity)
+    }
+
+    @After
+    fun finish() {
+        unmockkStatic(DogImageUrl::toDatabaseEntity)
+    }
+
+    @Test
+    fun `api service should be called when getting url`() {
+        val amountUrlsInDb = 1
```

Comment:
Minor: You can inline it because you are using it only in one place and it isn't important for this test